File: MATLAB/pve2sa.m
% PVE2SA(SX,H,THETA0) returns the prior variance of the additive effects on
% the outcome given the prior estimate of the proportion of variance
% explained by the additive effects (H). THETA0 is the (base 10) logarithm
% of the prior odds for inclusion, and SX is the sum of the sample variances
% for all the explanatory variables.
function sa = pve2sa (sx, h, theta0)
  sa = h./(1-h)./(sigmoid10(theta0) * sx);

File: unused-files/pve2sa.R
# Returns the prior variance of the regression coefficients given the
# prior estimate of the proportion of variance explained by the
# variables (h), and the prior log-odds that a variable is included in
# the model (logodds). Note that the base-10 logarithm is used here to
# define the prior log-odds.
pve2sa <- function (X, h, logodds) {
  sx <- sum(var1.cols(X))
  return(h/(1-h)/(sigmoid(log(10)*logodds)*sx))
}
